spec/requests/api/bookings_controller_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'rails_helper'
+
+RSpec.describe "Api::Bookings", type: :request do
+  
+  describe 'Create Booking' do
+    request_params_success_case = {
+      name: "user 1", 
+      email: "user1@mail.com", 
+      mobile: "98547854", 
+      booking_date: "2021/12/05", 
+      from_time:"12:00:00", 
+      to_time: "14:00:00"
+    }
+
+    request_params_failure_case = {
+      name: "user 1", 
+      email: "aaa@eer.kk", 
+      mobile: "", 
+      booking_date: "2021/12/05", 
+      from_time:"12:00:00", 
+      to_time: "14:00:00"
+    }
+    it "should return success status" do
+      post '/api/bookings/create', params: request_params_success_case, as: :json
+      expect(response).to have_http_status(200)
+    end
+
+    it "should have success message" do
+      post '/api/bookings/create', params: request_params_success_case, as: :json
+      json_response = JSON.parse(response.body)
+      expect(json_response['message']).to eq("Booking Created Successfully")
+    end
```

Comment:
This could be refactored as follows
```
context 'For success' do
  before do
    post '/api/bookings/create', params: request_params_success_case, as: :json
  end

  it 'returns success status' do
    expect(response).to have_http_status(200)
  end

   it 'returns success message' do
      json_response = JSON.parse(response.body)
      expect(json_response['message']).to eq("Booking Created Successfully")
  end
end
```

Reply:
Please refactor the Failure cases in same way.